include/modules/hyprland/workspaces.hpp

```diff
@@ -142,10 +142,23 @@ class Workspaces : public AModule, public EventHandler {
   void parse_config(const Json::Value& config);
   void register_ipc();
 
+  // workspace events
+  void on_workspace_activated(std::string const& payload);
+  void on_workspace_destroyed(std::string const& payload);
+  void on_workspace_created(std::string const& payload);
+  void on_workspace_moved(std::string const& payload);
+  void on_workspace_renamed(std::string const& payload);
```

Comment:
Thank you for doing this... I didn't like how busy that method was.

Reply:
This was also a hint by clang-tidy: it shows a warning if a function is too complex. Another reason to add it to the repo :)